### PR TITLE
Great cleanup!

### PR DESCRIPTION
**Pull request by sobolevn**
I've searched for duplicate definitions and removed most of them.
This was a manual job with some help of a new `stubtest`.

Rules:
- I haven't touched *any* magic methods (except a couple of identical `__init__`)
- I haven't touched any code with comments
- I haven't touched untyped code
- I haven't touched overloads, because they are too hard to read and verify
- I haven't touched any other advanced stuff

### CODE REVIEW

File: stdlib/xml/dom/minidom.pyi
Comment (AlexWaygood) [inline stdlib/xml/dom/minidom.pyi:56]: Can we maybe also leave the ones that are explicitly marked as `Incomplete` as they are? Since we've explicitly marked these as "we haven't figured out what they're meant to be yet", they might plausibly have different types in the subclass than in the superclass at runtime.

Comment (sobolevn) [reply stdlib/xml/dom/minidom.pyi:56]: I want to cancel all changes in this file :)
Because after my merge conflict I stopped understanding what is going on.

**Comment by sobolevn** [comment]
This error looks like a `stubtest` bug to me:

```
error: io.BufferedRandom.seek is inconsistent, stub argument "__offset" differs from runtime argument "target"
Stub: at line 65
def (self: io.IOBase, builtins.int, builtins.int =) -> builtins.int
Runtime:
def (self, target, whence=0, /)
```

Both types have pos-only parameters, but for some reason they are reported as incompatible.
I will restore these two with a comment for now.

**Comment by AlexWaygood** [comment]
> This error looks like a `stubtest` bug to me:

I believe it's actually stubtest working as intended; see https://github.com/python/mypy/issues/13113. I can see both sides of the argument here :-)

File: stdlib/importlib/abc.pyi
Comment (AlexWaygood) [inline stdlib/importlib/abc.pyi:45]: Given the comment here (this one doesn't actually exist on the class at runtime), maybe we should leave `exec_module` there in subclasses of `Loader`?

Comment (sobolevn) [reply stdlib/importlib/abc.pyi:45]: I don't really know the machinery in `importlib`, so I am fine with both.

Comment (AlexWaygood) [reply stdlib/importlib/abc.pyi:45]: Neither do I, so I'd prefer to play it safe and leave it in for now :)

File: stdlib/importlib/abc.pyi
Comment (AlexWaygood) [inline stdlib/importlib/abc.pyi:58]: Let's move this back to where it was before, to avoid noise in `git blame`

File: stdlib/importlib/machinery.pyi
Comment (AlexWaygood) [inline stdlib/importlib/machinery.pyi:148]: Same here

**Review by AlexWaygood — APPROVED** [review]
I haven't rigorously checked the whole thing to see if the signatures on subclasses are identical to the ones on superclasses, but I trust you to have done it properly :)

LGTM, though I'll wait to see if any other maintainers have any thoughts before merging.

**Comment by sobolevn** [comment]
> I trust you to have done it properly

I trust `stubtest` more than myself here 😆

**Comment by sobolevn** [comment]
Any other feedback? :)

I think that this PR can quickly have merge conflicts and this will require some time to re-think.

**Review by srittau — APPROVED** [review]
Not checked this in depth, either, but the passing tests give me confidence.

**Pull request merged**
